src/applications/disability-benefits/686c-674/config/helpers.js

```diff
@@ -71,3 +71,91 @@ export const cityTitle = (
     City <strong>or</strong> county
   </>
 );
+
+export const isInsideListLoopReturn = (
+  chapter,
+  outerField,
+  isInsideListLoop,
+  uiTitle,
+  formChapter,
+) => {
+  return {
+    'ui:title': uiTitle,
+    isOutsideUS: {
+      'ui:title': 'This occurred outside the US',
+    },
+    country: {
+      'ui:title': 'Country',
+      'ui:required': (formData, index) =>
+        formData[chapter][`${index}`][outerField]?.isOutsideUS,
+      'ui:options': {
+        hideIf: (formData, index) => {
+          if (!formData[chapter][`${index}`][outerField]?.isOutsideUS) {
+            return true;
+          }
+          return false;
+        },
+      },
+    },
+    state: {
+      'ui:title': 'State',
+      'ui:required': (formData, index) =>
+        !formData[chapter][`${index}`][outerField]?.isOutsideUS,
+      'ui:options': {
+        hideIf: (formData, index) => {
+          if (formData[chapter][`${index}`][outerField]?.isOutsideUS) {
+            return true;
+          }
+          return false;
+        },
+      },
+    },
+    city: {
+      'ui:required': formData => isChapterFieldRequired(formData, formChapter),
+      'ui:title': 'City',
+    },
+  };
+};
+
+export const isOutsideListLoopReturn = (
+  chapter,
+  outerField,
+  isInsideListLoop,
+  uiTitle,
+  formChapter,
+) => {
+  return {
+    'ui:title': 'Where were you married?',
+    isOutsideUS: {
+      'ui:title': 'This occurred outsite the US',
+    },
+    country: {
+      'ui:title': 'Country',
+      'ui:required': formData => formData[chapter][outerField]?.isOutsideUS,
+      'ui:options': {
+        hideIf: formData => {
+          if (!formData[chapter][outerField].isOutsideUS) {
+            return true;
+          }
+          return false;
+        },
+      },
+    },
+    state: {
+      'ui:title': 'State',
+      'ui:required': formData => !formData[chapter][outerField]?.isOutsideUS,
+      'ui:options': {
+        hideIf: formData => {
+          if (formData[chapter][outerField].isOutsideUS) {
+            return true;
+          }
+          return false;
```

Comment:
If `isOutsideUs` is a boolean like the name suggests, this could just be:

```js
return formData[chapter][outerField].isOutsideUS
```

If undefined values are a problem and boolean values are always needed here, then you could put a `!!` at the beginning to convert the value to a boolean.

Reply:
Yeah, I felt like writing it out like this is a little more explicit about what is going on here. I generally try to stay away from using `!!` because it can trip up a lot of less experienced engineers and tends to hide details of what is going on a bit.